File: apps/legacy_vault/legacy_vault/Core/Database/Schema.swift
import Foundation
import CoreData

// MARK: - CLCVoiceLogEntry (Core Data)

@objc(CLCVoiceLogEntry)
public final class CLCVoiceLogEntry: NSManagedObject {
    @NSManaged var id: String
    @NSManaged var title: String
    @NSManaged var recordingDate: Date
    @NSManaged var transcript: String
    @NSManaged var aiSummary: String?
    @NSManaged var aiEnrichment: String?
    @NSManaged var sentiment: Int16
    @NSManaged var durationMs: Int32
    @NSManaged var audioURL: String?
    @NSManaged var keywordsJSON: String
}

// Extension: VoiceLogEntry (Swift struct) typealias for fetch types
extension CLCVoiceLogEntry {
    /// Convenience typealias matching the Swift model name for fetch request type annotations.
    typealias SwiftModel = VoiceLogEntry
}

extension CLCInheritanceContact {
    static func fetch() -> NSFetchRequest<CLCInheritanceContact> {
        CLCInheritanceContact.fetchRequest() as! NSFetchRequest<CLCInheritanceContact>
    }
}

public extension CLCVaultRecord {
    static func fetch() -> NSFetchRequest<CLCVaultRecord> {
        CLCVaultRecord.fetchRequest() as! NSFetchRequest<CLCVaultRecord>
    }
}

extension CLCChatMessage {
    static func fetch() -> NSFetchRequest<CLCChatMessage> {
        CLCChatMessage.fetchRequest() as! NSFetchRequest<CLCChatMessage>
    }
}

extension CLCValueKeyword {
    static func fetch() -> NSFetchRequest<CLCValueKeyword> {
        CLCValueKeyword.fetchRequest() as! NSFetchRequest<CLCValueKeyword>
    }
}

// MARK: - CLCVaultRecord (Core Data)

@objc(CLCVaultRecord)
public final class CLCVaultRecord: NSManagedObject {
    @NSManaged var id: String
    @NSManaged var name: String
    @NSManaged var vaultType: String
    @NSManaged var encryptedData: Data
    @NSManaged var salt: Data
    @NSManaged var lastPingDate: Date
    @NSManaged var deadlineDays: Int32
    @NSManaged var targetEmails: String
    @NSManaged var status: String
}

// MARK: - CLCInheritanceContact (Core Data)

@objc(CLCInheritanceContact)
public final class CLCInheritanceContact: NSManagedObject {
    @NSManaged var id: String
    @NSManaged var name: String
    @NSManaged var email: String
    @NSManaged var relationship: String
    @NSManaged var notificationStatus: Int16
}

// MARK: - CLCChatMessage (Core Data)

@objc(CLCChatMessage)
public final class CLCChatMessage: NSManagedObject {
    @NSManaged var id: String
    @NSManaged var role: String
    @NSManaged var content: String
    @NSManaged var timestamp: Date
    @NSManaged var embeddingId: String?
}

// MARK: - CLCValueKeyword (Core Data)

@objc(CLCValueKeyword)
public final class CLCValueKeyword: NSManagedObject {
    @NSManaged var id: String
    @NSManaged var word: String
    @NSManaged var frequency: Int32
    @NSManaged var firstOccurrence: Date
    @NSManaged var lastOccurrence: Date
    @NSManaged var category: String
}

File: archive/legacy_quarantine/apps/legacy_vault_android/app/src/main/java/com/sfx/legacyvault/core/database/Schema.kt
package com.sfx.legacyvault.core.database

import androidx.room.Dao
import androidx.room.Entity
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.PrimaryKey
import androidx.room.Delete
import androidx.room.Query
import androidx.room.Update

@Entity(tableName = "voice_log_entry")
data class SchemaVoiceLogEntry(
    @PrimaryKey val id: String = "",
    val title: String = "",
    val recordingDate: Long = 0L,
    val transcript: String = "",
    val aiSummary: String? = null,
    val aiEnrichment: String? = null,
    val sentiment: Int = 0,
    val durationMs: Int = 0,
    val keywords: String? = null,
    val embeddingId: String? = null
)

@Entity(tableName = "vault_record")
data class SchemaVaultRecord(
    @PrimaryKey val id: String = "",
    val name: String = "",
    val vaultType: String = "",
    val encryptedData: String = "",
    val salt: String = "",
    val lastPingDate: Long = 0L,
    val deadlineDays: Int = 0,
    val targetEmails: String? = null,
    val status: String = "active"
)

@Entity(tableName = "inheritance_contact")
data class SchemaInheritanceContact(
    @PrimaryKey val id: String = "",
    val name: String = "",
    val email: String = "",
    val relationship: String = "",
    val notificationStatus: Int = 0
)

@Entity(tableName = "chat_message")
data class SchemaChatMessage(
    @PrimaryKey val id: String = "",
    val role: String = "",
    val content: String = "",
    val timestamp: Long = 0L,
    val embeddingId: String? = null
)

@Entity(tableName = "value_keyword")
data class SchemaValueKeyword(
    @PrimaryKey val id: String = "",
    val word: String = "",
    val frequency: Int = 0,
    val firstOccurrence: Long = 0L,
    val lastOccurrence: Long = 0L,
    val category: String = ""
)

@Dao
interface VoiceLogEntryDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(entry: SchemaVoiceLogEntry): Long

    @Update
    suspend fun update(entry: SchemaVoiceLogEntry)

    @Query("SELECT * FROM voice_log_entry ORDER BY recordingDate DESC")
    suspend fun getAll(): List<SchemaVoiceLogEntry>

    @Query("SELECT * FROM voice_log_entry WHERE id = :id")
    suspend fun getById(id: String): SchemaVoiceLogEntry?

    @Query("DELETE FROM voice_log_entry")
    suspend fun deleteAll()
}

@Dao
interface VaultRecordDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(record: SchemaVaultRecord): Long

    @Update
    suspend fun update(record: SchemaVaultRecord)

    @Query("SELECT * FROM vault_record ORDER BY lastPingDate DESC")
    suspend fun getAll(): List<SchemaVaultRecord>

    @Query("SELECT * FROM vault_record WHERE id = :id")
    suspend fun getById(id: String): SchemaVaultRecord?

    @Query("UPDATE vault_record SET lastPingDate = :pingTimestamp, status = 'active' WHERE id = :id")
    suspend fun ping(id: String, pingTimestamp: Long)

    @Query("DELETE FROM vault_record")
    suspend fun deleteAll()
}

@Dao
interface InheritanceContactDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(contact: SchemaInheritanceContact): Long

    @Update
    suspend fun update(contact: SchemaInheritanceContact)

    @Query("SELECT * FROM inheritance_contact ORDER BY relationship")
    suspend fun getAll(): List<SchemaInheritanceContact>

    @Query("SELECT * FROM inheritance_contact WHERE id = :id")
    suspend fun getById(id: String): SchemaInheritanceContact?

    @Query("DELETE FROM inheritance_contact")
    suspend fun deleteAll()
}

@Dao
interface ChatMessageDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(message: SchemaChatMessage): Long

    @Update
    suspend fun update(message: SchemaChatMessage)

    @Query("SELECT * FROM chat_message ORDER BY timestamp DESC")
    suspend fun getAll(): List<SchemaChatMessage>

    @Query("SELECT * FROM chat_message WHERE id = :id")
    suspend fun getById(id: String): SchemaChatMessage?

    @Query("DELETE FROM chat_message")
    suspend fun deleteAll()
}

@Dao
interface ValueKeywordDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(keyword: SchemaValueKeyword): Long

    @Update
    suspend fun update(keyword: SchemaValueKeyword)

    @Query("SELECT * FROM value_keyword ORDER BY frequency DESC")
    suspend fun getAll(): List<SchemaValueKeyword>

    @Query("SELECT * FROM value_keyword WHERE id = :id")
    suspend fun getById(id: String): SchemaValueKeyword?

    @Query("DELETE FROM value_keyword")
    suspend fun deleteAll()
}
